src/Microsoft.Extensions.Caching.Abstractions/IMemoryCache.cs

```diff
@@ -30,5 +30,11 @@ public interface IMemoryCache : IDisposable
         /// </summary>
         /// <param name="key">An object identifying the entry.</param>
         void Remove(object key);
+
+        /// <summary>
+        /// Attempt to remove the given percentage of the total entries in the cache.
+        /// </summary>
+        /// <param name="percentage">The percentage of cache entries to remove.</param>
+        void Compact(double percentage);
```

Comment:
> The point of using percentages is that they don't need to know the size

I would say that knowing what percentage you need is also improbable. I would just provide a strategy instead. Like FIFO or based on usage. This strategy would then have access to all the calls to track what it needs (count, memory, usage frequency of each entry, ...). This strategy would be responsible to call Compact internally based on its own logic. And if I want a strategy that is dumb and just removes a percentage of random items then I could.